src/eventListeners/onImageRenderedBrushEventHandler.js

```diff
@@ -142,9 +145,533 @@ function renderInactiveLabelMaps(
+      labelmap2D,
+      isActiveLabelMap
+    );
+  }
+
+  if (configuration.renderOutline) {
+    renderOutline(evt, labelmap3D, labelmapIndex, labelmap2D, isActiveLabelMap);
+  }
+}
+
+/**
+ * RenderOutline - Renders the outlines of segments to the canvas.
+ *
+ * @param  {Object} evt             The cornerstone event.
+ * @param  {Object} labelmap3D      The 3D labelmap.
+ * @param  {number} labelmapIndex   The index of the labelmap.
+ * @param  {Object} labelmap2D      The 2D labelmap for this current image.
+ * @param  {number} isActiveLabelMap   Whether the labelmap is active.
+ * @returns {null}
+ */
+function renderOutline(
+  evt,
+  labelmap3D,
+  labelmapIndex,
+  labelmap2D,
+  isActiveLabelMap = true
+) {
+  // Don't bother rendering a whole labelmap with full transparency!
+  if (isActiveLabelMap && configuration.outlineAlpha === 0) {
+    return;
+  } else if (!isActiveLabelMap && configuration.outlineAlphaInactive === 0) {
+    return;
+  }
+
+  const eventData = evt.detail;
+  const { element, canvasContext } = eventData;
+
+  const lineWidth = configuration.outlineWidth || 1;
+  const lineSegments = getLineSegments(
+    eventData,
+    labelmap3D,
+    labelmap2D,
+    lineWidth
+  );
+
+  const context = getNewContext(canvasContext.canvas);
+  const colorMapId = `${state.colorMapId}_${labelmapIndex}`;
+  const colorLutTable = state.colorLutTables[colorMapId];
+
+  const previousAlpha = context.globalAlpha;
+
+  context.globalAlpha = isActiveLabelMap
+    ? configuration.outlineAlpha
+    : configuration.outlineAlphaInactive;
+
+  // Draw outlines.
+  draw(context, context => {
+    for (let i = 0; i < lineSegments.length; i++) {
+      if (lineSegments[i]) {
+        const color = colorLutTable[i];
+
+        drawLines(
+          context,
+          element,
+          lineSegments[i],
+          {
+            color: `rgba(${color[0]}, ${color[1]}, ${color[2]}, 1.0 )`,
+            lineWidth,
+          },
+          'canvas'
+        );
+      }
+    }
+  });
+
+  context.globalAlpha = previousAlpha;
+}
+
+/**
+ * GetLineSegments - Returns an object containing all the line segments to be
+ * drawn the canvas.
+ *
+ * @param  {Object} eventData The eventdata associated with the cornerstone event.
+ * @param  {Object} labelmap3D The 3D labelmap.
+ * @param  {Object} labelmap2D The 2D labelmap for this current image.
+ * @param  {number} lineWidth The width of the line segments.
+ *
+ * @returns {Object[][]} An array of arrays of lines for each segment.
+ */
+function getLineSegments(eventData, labelmap3D, labelmap2D, lineWidth) {
+  const { element, image } = eventData;
+  const cols = image.width;
+  const rows = image.height;
+
+  const pixelData = labelmap2D.pixelData;
+  const activeSegmentIndex = labelmap3D.activeSegmentIndex;
+  const lineSegments = [];
+
+  labelmap2D.segmentsOnLabelmap.forEach(segmentIndex => {
+    lineSegments[segmentIndex] = [];
+  });
+
+  // TEMP - Do this in a cleaner way.
```

Comment:
Not sure if this _is_ the right way for now. `segmentsOnLabelmap` is set at the end of a brush stroke, so you need to render its outline before its set. For context, the intent of `segmentsOnLabelmap` is to have a quick way to check what memory needs to be allocated for things like DICOM seg export.

Reply:
I don't think anyone is going to export mid brush stroke so I guess that's fine.